internal/util/profiles.go

```diff
@@ -0,0 +1,124 @@
+package util
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type Result = map[string]interface{}
+
+var profileFilePath = "/run/suseconnect/"
+
+// Profile struct is for handling lage data blobs
+// ProfileID: an index to reference ProfileData, at the moment
+//
+//	This will be a sha256 caluculated from ProfileData
+//	since, ProfileData can be any type (string, array
+//	of strings, etc) it may need processing before the
+//	sha256 can be calcuated
+//
+// ProfileData: can be any data such array of strings, a single large string
+type Profile struct {
+	ProfileID   string `json:"profileId"`   // Sha256 of ProfileData
+	ProfileData any    `json:"profileData"` // Data associated with ProfileI
+}
+
+func UpdateCache(updateCache bool, tag string, cacheFile string, dataBlock any) (Result, error) {
+
+	var hashInput = ""
+	switch v := dataBlock.(type) {
```

Comment:
As detailed before, let it be a string and avoid type casting shenanigans.

Reply:
Just saying all of this because I don't think we need to use `any` unless there's an extremely good argument on using it.